article.md

```diff
@@ -0,0 +1,333 @@
+        "Invalid signature length. Signature should have exactly 5 elements."
+    ){
+        assert signature_len = 5;
+    }
+
+    let v = signature[0];
+    let r = Uint256(signature[2], signature[1]);
+    let s = Uint256(signature[4], signature[3]);
+    
+    let (local keccak_ptr_start) = alloc();
+    let keccak_ptr = keccak_ptr_start;
+
+    let (hash_uint) = get_hash{keccak_ptr=keccak_ptr}(starknet_address, domain_hash);
+   
+    verify_eth_signature_uint256{keccak_ptr=keccak_ptr}(hash_uint, r, s, v, eth_address);
+
+    finalize_keccak(keccak_ptr_start=keccak_ptr_start, keccak_ptr_end=keccak_ptr);
+
+    return ();
+}
+```
+
+## Signing with an Ethereum wallet
+Let's use Metamask as an example. You just need to call the `signTypedData_v4` function (inputting the correct address in the `myStarkNetAddress` field) and it will return the signature that we need.<br/>
+```js
+const msg = {
+  domain: {
+    name: "Confirm the ownership of this StarkNet address",
+    verifyingContract: '0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF',
+    version: '1',
+  },
+  message: {
+    "starknetAddress": 12356789,
+  },
+  primaryType: 'Payload',
+  types: {
+    domain: [
+      { name: 'name', type: 'string' },
+      { name: 'verifyingContract', type: 'address' },
+      { name: 'version', type: 'string' },
+    ],
+    Payload: [
+      { name: 'starknetAddress', type: 'uint256' },
+    ],
+  },
+};
+const accounts = await window.ethereum.request({method: "eth_requestAccounts"})
+await window.ethereum.request({
+    method: "eth_signTypedData_v4",
+    from: accounts[0],
+    params: [accounts[0], JSON.stringify(msg)]
+})
+```
+After you input this JS code into the console on the MetaMask website it will trigger a signing request (of course normally you wouldn't do it this way but it's just for demonstration's sake).<br/><br/>
+<img alt="Screenshot 2022-10-12 at 10 41 54" src="./graphics/metamask.png"><br/><br/>
+As you can see everything is transparent so you don't have to sign something blindly - that's one of the pros of using EIP-712 standard.<br/><br/>
+<img alt="Screenshot 2022-10-12 at 10 41 54" src="./graphics/signature.png"><br/>
+When you sign this message you will receive the message signature that we need. Then you would need to call the contract that we're writing from your StarkNet account which expects two input variables: your Ethereum address and the message signature that you have just created.
+## Recreating and hashing the message
+### The basic idea
+Since the ownership of the Ethereum account is confirmed by signing the message for which the private key is needed, we also need to confirm that the user owns the StarkNet account of which's address he is using. That's why we're deliberately not passing `starknet_address` as an input variable, instead, we call `get_caller_address()`, which we'll use as the address we need. Connection will be created only if the address of an account's executing the tx is the one signed.
```

Comment:
"StarkNet account that is supposed to be connected".

"The connection will be created.. of the account..."